src/cli/auto_update.rs

```diff
@@ -228,11 +235,19 @@ async fn update_repo(repo: &RemoteRepo, bleeding_edge: bool, freeze: bool) -> Re
     let output = cmd.output().await?;
     let mut rev = if output.status.success() {
         let rev = String::from_utf8_lossy(&output.stdout).trim().to_string();
-        let rev = get_best_candidate_tag(tmp_dir.path(), &rev, &repo.rev)
-            .await
-            .unwrap_or(rev);
-        trace!("Using best candidate tag `{rev}`");
-        rev
+        if let Some(best) =
+            get_best_candidate_tag(tmp_dir.path(), &rev, &repo.rev, cooldown_days).await?
```

Comment:
I think this implementation isn’t quite right. Right now, auto‑update works by grabbing the latest tag that’s closest to `FETCH_HEAD`, and `get_best_candidate_tag` just picks the best tag name if there are multiple tags point to the same commit.

But for cooldown support, we can’t keep using “the latest tag near FETCH_HEAD.” Instead, we should list all the tags, sort them by creation time, and pick the newest one that’s still older than the cooldown period.